### PR TITLE
fix(default_adapi): subscribe planning factor

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

New factor msgs were defined in https://github.com/tier4/tier4_autoware_msgs/pull/155 to output various information of module running state. And, each planning module will use them.

In this PR, I fixed adapi node to convert msg type from `tier4_planning_msgs/PlanningFactorArray` to `autoware_adapi_v1_msgs/VelocityFactorArray(SteeringFactorArray)`.

## Related links

**Parent Issue:**

- https://github.com/autowarefoundation/autoware.universe/pull/9676
- https://github.com/tier4/tier4_autoware_msgs/pull/155

## How was this PR tested?

image

1. launch psim.
2. put ego and goal.
3. put object on ego path.
4. check the topic `/api/planning/velocity_factors` by echo command.
5. confirm that the node outputs factor like following one.

```
header:
  stamp:
    sec: 1735008280
    nanosec: 610628888
  frame_id: map
factors:
- pose:
    position:
      x: 61578.199978951016
      y: 56213.78579257302
      z: 78.16299873101634
    orientation:
      x: -3.6191521078641166e-05
      y: -0.0006830251435586911
      z: -0.052912858993544
      w: 0.9985988992182779
  distance: 14.520707130432129
  status: 1
  behavior: route-obstacle
  sequence: ''
  detail: ''
  cooperation: []
```

## Notes for reviewers

None.

## Interface changes

| Version | Topic Type      | Topic Name        | Message Type        | Description       |
|:--------|:----------------|:------------------|:--------------------|:------------------|
| Old     | Sub | `/planing/planning_factors/xxx` | `tier4_planning_msgs/PlanningFactorArray` | Topic description |
| New     | Sub | `/planing/velocity(steering)_factors/xxx` | `autoware_adapi_v1_msgs/VelocityFactorArray(SteeringFactorArray)` | Topic description |

## Effects on system behavior

None.

### CODE REVIEW

File: system/autoware_default_adapi/src/planning.hpp [hunk 1]
```diff
@@ -30,22 +35,53 @@
...
+  {"obstacle_cruise_planner", PlanningBehavior::ROUTE_OBSTACLE},
+  {"motion_velocity_planner", PlanningBehavior::ROUTE_OBSTACLE},
+  {"walkway", PlanningBehavior::SIDEWALK},
+  {"start_planner", PlanningBehavior::START_PLANNER},
+  {"stop_line", PlanningBehavior::STOP_SIGN},
+  {"surround_obstacle_checker", PlanningBehavior::SURROUNDING_OBSTACLE},
+  {"traffic_light", PlanningBehavior::TRAFFIC_SIGNAL},
+  {"detection_area", PlanningBehavior::USER_DEFINED_DETECTION_AREA},
+  {"virtual_traffic_light", PlanningBehavior::VIRTUAL_TRAFFIC_LIGHT},
+  {"run_out", PlanningBehavior::RUN_OUT}};
```
Comment (isamu-takagi) [inline system/autoware_default_adapi/src/planning.hpp:71]: These variables can be moved to cpp file.

File: system/autoware_default_adapi/src/planning.cpp [hunk 2]
```diff
@@ -49,16 +49,117 @@ std::vector<typename rclcpp::Subscription<T>::SharedPtr> init_factors(
...
+    }
+
+    if (factor.control_points.size() < 2) {
+      continue;
+    }
+
+    if (conversion_map.count(factor.module) == 0) {
+      continue;
+    }
+
```
Comment (isamu-takagi) [inline system/autoware_default_adapi/src/planning.cpp:108]: Please also add a check for `direction_map` to avoid an exception by the `at` function.

**Review by isamu-takagi — COMMENTED** [review]
The velocity/steering factor has a status, but planning factor did not. The status of velocity factor will be automatically completed. However, the status of steering factor will not be assigned.

File: system/autoware_default_adapi/src/planning.cpp (hunk 2 above)
Comment (satoshi-ota) [reply system/autoware_default_adapi/src/planning.cpp:136]: Fixed in https://github.com/tier4/autoware.universe/pull/1734/commits/615d1bb7371048c3c0bf5a71f067a92e378c8d9e.

**Comment by satoshi-ota** [comment]
> The velocity/steering factor has a status, but planning factor does not. The status of velocity factor will be automatically completed. However, the status of steering factor will not be assigned.

Thanks for your comment. You're right.

I fixed logic in https://github.com/tier4/autoware.universe/pull/1734/commits/5d55a4b03f37e16e0e23d14a02e08c272cbf0022 to fill status automatically.

status: `SteeringFactor::APPROACHING`
Screenshot from 2024-12-25 09-26-08

status: `SteeringFactor::TURNING`
Screenshot from 2024-12-25 09-26-27

File: system/autoware_default_adapi/src/planning.hpp (hunk 1 above)
Comment (satoshi-ota) [reply system/autoware_default_adapi/src/planning.hpp]: Fixed in https://github.com/tier4/autoware.universe/pull/1734/commits/615d1bb7371048c3c0bf5a71f067a92e378c8d9e.

**Review by isamu-takagi — APPROVED** [review]
LGTM

**Pull request merged**